<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Discovery Only</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Discovery Only</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(MultiStudyNMF)</span></code></pre></div>
<p>We provide demo data for this example as <code>data(lung)</code>. The
R chunk below shows how this data was prepared, using data from PCAWG
that can be downloaded <a href="https://www.synapse.org/Synapse:syn11726620">here</a>. As a simple
example, we considered two studies: lung adenocarcinoma and lung
squamous cell carcinoma. We ordered the mutational categories of this
data to match that of the COSMIC reference signatures, for easier
comparison later on.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="do">## NOT RUN</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co"># Read in data and re-order mutational categories to match COSMIC</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;../../../Downloads/WGS_PCAWG_2018_02_09/WGS_PCAWG.96.csv&#39;</span>)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">data</span>(cosmic)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">substr</span>(data<span class="sc">$</span>Trinucleotide,<span class="dv">0</span>,<span class="dv">1</span>),<span class="st">&#39;[&#39;</span>,</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>                         data<span class="sc">$</span>Mutation.type,<span class="st">&#39;]&#39;</span>,</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>                         <span class="fu">substr</span>(data<span class="sc">$</span>Trinucleotide,<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[cosmic<span class="sc">$</span>Type,]</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co"># Subset to the two cancers of interest</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>M <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">as.matrix</span>(data[,<span class="fu">substr</span>(<span class="fu">colnames</span>(data),<span class="dv">0</span>,<span class="dv">10</span>)<span class="sc">==</span><span class="st">&#39;Lung.Adeno&#39;</span>]),</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>          <span class="fu">as.matrix</span>(data[,<span class="fu">substr</span>(<span class="fu">colnames</span>(data),<span class="dv">0</span>,<span class="dv">10</span>)<span class="sc">==</span><span class="st">&#39;Lung.SCC..&#39;</span>]))</span></code></pre></div>
<p>Note that the input data is a list of two matrices, each in
mutational motifs-by-samples format. These correspond to the lung
adenocarcinoma data (38 samples) and the lung squamous cell carcinoma
data (48 samples) respectively.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">data</span>(lung)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>M <span class="ot">&lt;-</span> lung<span class="sc">$</span>M</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="fu">str</span>(M)</span></code></pre></div>
<pre><code>## List of 2
##  $ : int [1:96, 1:38] 2412 1285 454 1807 579 239 87 393 724 314 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:96] &quot;A[C&gt;A]A&quot; &quot;A[C&gt;A]C&quot; &quot;A[C&gt;A]G&quot; &quot;A[C&gt;A]T&quot; ...
##   .. ..$ : chr [1:38] &quot;Lung.AdenoCA..SP53460&quot; &quot;Lung.AdenoCA..SP50827&quot; &quot;Lung.AdenoCA..SP50263&quot; &quot;Lung.AdenoCA..SP50317&quot; ...
##  $ : int [1:96, 1:48] 1404 758 349 848 378 184 75 312 594 231 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:96] &quot;A[C&gt;A]A&quot; &quot;A[C&gt;A]C&quot; &quot;A[C&gt;A]G&quot; &quot;A[C&gt;A]T&quot; ...
##   .. ..$ : chr [1:48] &quot;Lung.SCC..SP56638&quot; &quot;Lung.SCC..SP56537&quot; &quot;Lung.SCC..SP57586&quot; &quot;Lung.SCC..SP56541&quot; ...</code></pre>
<p>Now we run the discovery-only sampler.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">62524</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">discovery_fit</span>(M)</span></code></pre></div>
<p>We can view the signature sharing matrix, which describes which of
the estimated signatures are present in each of the two studies.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>out<span class="sc">$</span>A</span></code></pre></div>
<pre><code>##        sig1 sig2 sig3 sig4 sig5 sig6 sig7 sig8 sig9 sig10 sig11 sig12 sig13 sig14
## study1    1    1    0    1    1    1    1    0    1     1     1     0     1     1
## study2    1    1    1    1    1    1    1    1    1     1     1     1     1     1</code></pre>
<p>Here, we see that we found a total of 14 signatures, of which the
majority belong to both studies. There are three signatures (labeled as
sig3, sig8, and sig12) which are only found in the second study,
i.e. lung squamous cell carcinoma. This output also contains the
signatures estimates (<code>out$P</code>), the exposures estimates
(<code>out$E</code>), and the study-specific normalization factors
(<code>out$W</code>). More detail about this output can be found in the
documentation by viewing <code>?discovery_fit</code>.</p>
<p>Below, we provide two convenience functions to help interpret the
output. First, assuming that the mutational motifs of the input data
matched the order of the COSMIC signatures, we can annotate each
signature by its closest match to the COSMIC v3.2 signatures (further
details in <code>?label_A</code>) and view the signature sharing matrix
as follows:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">label_A</span>(out<span class="sc">$</span>A,out<span class="sc">$</span>P)</span></code></pre></div>
<pre><code>##        Pseudo SBS18 Pseudo SBS3 Pseudo SBS33 Pseudo SBS13 Pseudo SBS5 Pseudo SBS1 Pseudo SBS17b
## study1            1           1            0            1           1           1             1
## study2            1           1            1            1           1           1             1
##        Pseudo SBS3.1 SBS2 SBS4 Pseudo SBS3.2 Pseudo SBS92 Pseudo SBS40 Pseudo SBS25
## study1             0    1    1             1            0            1            1
## study2             1    1    1             1            1            1            1</code></pre>
<p>Some signatures may only be present in a small subset of samples. We
can view the proportion of samples in each study with relative exposures
over a pre-specified threshold as follows:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">proportion_A</span>(out,<span class="at">thresh=</span><span class="fl">0.05</span>,<span class="at">label=</span>T)</span></code></pre></div>
<pre><code>##        Pseudo SBS18 Pseudo SBS3 Pseudo SBS33 Pseudo SBS13 Pseudo SBS5 Pseudo SBS1 Pseudo SBS17b
## study1    0.4736842   0.2368421       0.0000    0.4473684   0.2631579   0.4736842     0.1052632
## study2    0.4583333   0.2708333       0.0625    0.7083333   0.4583333   0.1041667     0.0000000
##        Pseudo SBS3.1      SBS2      SBS4 Pseudo SBS3.2 Pseudo SBS92 Pseudo SBS40 Pseudo SBS25
## study1         0.000 0.3684211 0.7631579     0.4210526        0.000    0.7105263    0.6578947
## study2         0.125 0.1458333 0.9583333     0.1041667        0.375    0.9583333    0.8958333</code></pre>
<p>For example, here we see that one of the signatures unique to
squamous cell carcinoma, SBS33, only has high exposures in a small
subset (around 6%) of samples, whereas another unique signature, SBS92,
has high exposures to a larger subset (38%).</p>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
